drivers/watchdog/wdt_mcux_imx_wdog.c

```diff
@@ -42,10 +42,10 @@ static int mcux_wdog_setup(const struct device *dev, uint8_t options)
 	}
 
 	data->wdog_config.workMode.enableStop =
-		(options & WDT_OPT_PAUSE_IN_SLEEP) == 0U;
+		(options & WDT_OPT_PAUSE_IN_SLEEP) == WDT_OPT_PAUSE_IN_SLEEP;
 
 	data->wdog_config.workMode.enableDebug =
```

Comment:
So the documentation in the iMX RT1024 RM says the following for the watchdog WDBG bit says the following:

> The WDOG timer can be configured for continual operation, or for suspension during
debug mode. If the WDOG debug enable (WDBG) bit is set in the Watchdog Control
(WCR), the WDOG timer operation is suspended in debug mode.

Setting the `enableDebug` field to true for the MCUX `wdog01` driver will result in the `WDBG` bit being cleared: https://github.com/zephyrproject-rtos/hal_nxp/blob/9702923eeb6f4a9ca063ca3b200324b118d3e843/mcux/mcux-sdk/drivers/wdog01/fsl_wdog.c#L112

So it seems that setting the `enableDebug` field for the MCUX driver will cause the watchdog to continue running in debug mode. Based on the documentation of the `WDT_OPT_PAUSE_HALTED_BY_DBG` bit: https://github.com/zephyrproject-rtos/zephyr/blob/c82799bd4c2980f579afcb0b3dc07991f765579d/include/zephyr/drivers/watchdog.h#L38

It seems like the current behavior is correct, right?

Reply:
ping @VinFar

Reply:
Sorry for the delay. The bug wax fixed in Zephyr 3.7.0 and in a different place than I was looking, so I just overlooked it.
It was fixed in NXP SDK 2.15.0 which is used in zephyr 3.7

However, it still persists in older versions.

See this version of the file in zephyr 3.6.0:
https://github.com/zephyrproject-rtos/hal_nxp/blob/d45b14c198d778658b7853b48378d2e132a6c4be/mcux/mcux-sdk/drivers/wdog01/fsl_wdog.c#L112

Should this be backported?

Reply:
Sorry for the delay. The bug wax fixed in Zephyr 3.7.0 and in a different place than I was looking, so I just overlooked it.
It was fixed in NXP SDK 2.15.0 which is used in zephyr 3.7

However, it still persists in older versions.

See this version of the file in zephyr 3.6.0:
https://github.com/zephyrproject-rtos/hal_nxp/blob/d45b14c198d778658b7853b48378d2e132a6c4be/mcux/mcux-sdk/drivers/wdog01/fsl_wdog.c#L112

Should this be backported?

Reply:
Ah, this is fun. The fix for the issue was effectively the SDK 2.15 update, but we cannot backport that. I *think* the best approach here would be the following:
- update the issue text to explicitly refer to 3.6, and highlight how the issue has been fixed in newer releases by the SDK update
- Create a PR targeting the 3.6 branch (and be explicit in the description as to how this issue only exists in 3.6, and not on main)

I suppose the other option we could pursue would be patching the 2.14 SDK watchdog driver, and syncing that change back to 3.6. I'd rather just change the logic within the Zephyr driver though, since that reduces the number of changes needed to fix this issue

Reply:
I'm not sure it is worth going back to v3.6 to address as support for v3.6 ends next month.